src/simplify_algebra.cpp

```diff
@@ -725,6 +725,54 @@ struct find_concat_op
     }
 };
 
+struct find_concat_conv
+{
+    auto matcher() const
+    {
+        return match::name("concat")(
+            match::all_of[match::inputs()](match::used_once(), match::name("convolution")));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto ins  = r.result;
+        auto axis = ins->get_operator().to_value()["axis"].to<int>();
+        if(axis != 1)
+            return;
+        if(ins->inputs().empty())
+            return;
```

Comment:
Would this case occure at all ? All of the inputs to the concat must be convolution. It can't be empty